pulsar-broker/src/main/java/org/apache/pulsar/broker/s3offload/impl/BlockAwareSegmentInputStreamImpl.java

```diff
@@ -209,5 +209,14 @@ public int getBlockEntryBytesCount() {
         return BLOCK_END_PADDING;
     }
 
+    // Get the block size after uploaded `bytesUploaded` bytes
+    public static int getBlockSize(ReadHandle readHandle, int maxBlockSize, long startEntryId, long bytesUploaded) {
```

Comment:
rename a few things here:

getBlockSize -> calculateBlockSize
startEntryId -> firstEntryToWrite
bytesUploaded -> entryBytesAlreadyWritten

Also, maxBlockSize should be the first argument, as the other three are ledger related, so should be grouped together.